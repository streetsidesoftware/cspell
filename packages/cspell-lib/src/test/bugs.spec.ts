import * as fsp from 'fs/promises';
import * as path from 'path';

import * as cspell from '../index';

const samples = path.join(__dirname, '..', '..', 'samples', 'bug-fixes');
const configFile = path.join(samples, 'cspell.json');

const files = ['bug345.ts', '../src/sample.go'];

describe('Validate Against Bug Fixes', () => {
    jest.setTimeout(10000);
    function t(filename: string) {
        test(`validate ${filename}`, async () => {
            const fullFilename = path.resolve(samples, filename);
            const ext = path.extname(filename);
            const text = await fsp.readFile(fullFilename, 'utf-8');
            const languageIds = cspell.getLanguagesForExt(ext);
            const settings = cspell.mergeSettings(cspell.getDefaultBundledSettings(), cspell.readSettings(configFile));
            const fileSettings = cspell.combineTextAndLanguageSettings(settings, text, languageIds);
            const result = await cspell.validateText(text, fileSettings);
            expect(result).toMatchSnapshot();
        });
    }

    files.forEach(t);
});
