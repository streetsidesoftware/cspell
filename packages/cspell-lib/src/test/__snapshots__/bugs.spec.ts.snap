// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validate Against Bug Fixes validate ../src/sample.go 1`] = `
[
  {
    "isFlagged": false,
    "isFound": false,
    "line": {
      "offset": 265,
      "text": "	longname(i)
",
    },
    "offset": 266,
    "text": "longname",
  },
  {
    "isFlagged": false,
    "isFound": false,
    "line": {
      "offset": 344,
      "text": "func longname(i I) {
",
    },
    "offset": 349,
    "text": "longname",
  },
  {
    "isFlagged": false,
    "isFound": false,
    "line": {
      "offset": 400,
      "text": "/* Deliberate misspelling: garbbage. */
",
    },
    "offset": 427,
    "text": "garbbage",
  },
]
`;

exports[`Validate Against Bug Fixes validate bug345.ts 1`] = `[]`;
